sdk/ml/azure-ai-ml/tests/component/unittests/test_data_transfer_component_entity.py

```diff
@@ -0,0 +1,139 @@
+        assert isinstance(component_entity, DataTransferImportComponent)
+
+        data_transfer_copy_component = DataTransferImportComponent(
+            task=DataTransferTaskType.IMPORT_DATA,
+            name="datatransfer_s3_blob",
+            display_name="Data Transfer s3-blob",
+            source={"type": "file_system"},
+            outputs={"sink": {"type": "uri_folder"}},
+            base_path="./tests/test_configs/components/data_transfer",
+        )
+
+        omit_fields = ["name", "id", "$schema"]
+        yaml_dict = pydash.omit(dict(component_entity._to_dict()), *omit_fields)
+        sdk_dict = pydash.omit(dict(data_transfer_copy_component._to_dict()), *omit_fields)
+        assert sdk_dict == yaml_dict
+
+    def test_serialize_deserialize_export_task_component(self, mock_machinelearning_client: MLClient):
+        test_path = "./tests/test_configs/components/data_transfer/export_blob_to_database.yaml"
+        component_entity = load_component_entity_from_yaml(test_path, mock_machinelearning_client,
+                                                           _type="data_transfer")
+        assert isinstance(component_entity, DataTransferExportComponent)
+
+        data_transfer_copy_component = DataTransferExportComponent(
+            task=DataTransferTaskType.EXPORT_DATA,
+            name="datatransfer_blob_azuresql",
+            display_name="Data Transfer blob-azuresql",
+            inputs={"source": {"type": "uri_folder"}},
+            sink={"type": "database"},
+            base_path="./tests/test_configs/components/data_transfer",
+        )
+
+        omit_fields = ["name", "id", "$schema"]
+        yaml_dict = pydash.omit(dict(component_entity._to_dict()), *omit_fields)
+        sdk_dict = pydash.omit(dict(data_transfer_copy_component._to_dict()), *omit_fields)
+        assert sdk_dict == yaml_dict
+
+    def test_copy_task_component_entity(self):
+        component = DataTransferCopyComponent(
+            task=DataTransferTaskType.COPY_DATA,
+            name="datatransfer_copy_files",
+            display_name="Data Transfer Component copy-files",
+            inputs={
+                "folder1": {"type": "uri_folder"},
+            },
+            outputs={
+                "output_folder": {"type": "uri_folder"},
+            },
+            data_copy_mode= DataCopyMode.MERGE_WITH_OVERWRITE,
+            base_path="./tests/test_configs/components/data_transfer",
+        )
+        omit_fields = [
+            "properties.component_spec.$schema",
+            "properties.component_spec._source",
+        ]
+        component._validate()
+        component_dict = component._to_rest_object().as_dict()
+        component_dict = pydash.omit(component_dict, *omit_fields)
+
+        yaml_path = "./tests/test_configs/components/data_transfer/copy_files.yaml"
+        yaml_component = load_component(yaml_path)
+        yaml_component_dict = yaml_component._to_rest_object().as_dict()
```

Comment:
May add some more invalid cases testing in next PR

Reply:
will added in next pr